### PR TITLE
admin router

### PR DESCRIPTION
**Pull request by tjstebbing**
initial setup of stuff we need to build the router, ie: ports and webUIs on state

### CODE REVIEW

File: pkg/pups.go [hunk 1]
```diff
@@ -50,8 +50,17 @@ type PupState struct {
...
 }
 
+// Represents a Web UI exposed port from the manifest
+type PupWebUI struct {
+	Name     string `json:"name"`
+	Internal int    `json:"-"`
+	Port     int    `json:"port"`
+}
+
+// noodles ?
```
Comment (SomeoneWeird) [inline pkg/pups.go:63]: noodles?

**Review by SomeoneWeird — COMMENTED** [review]
So this code looks good. The missing part is now us opening firewall entries for these dynamically allocated ports.. which require doing a nix rebuild. Thoughts?

File: pkg/pups.go (hunk 1 above)
Comment (tjstebbing) [reply pkg/pups.go:63]: umm

**Comment by tjstebbing** [comment]
> So this code looks good. The missing part is now us opening firewall entries for these dynamically allocated ports.. which require doing a nix rebuild. Thoughts?

They are exposed by dogeboxd? they are not in a pup container, this already works in my vm ?

**Comment by tjstebbing** [comment]
Added opening firewall ports for admin ports

**Review by SomeoneWeird — APPROVED** [review]

**Pull request merged**